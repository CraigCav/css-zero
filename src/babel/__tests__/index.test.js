const path = require('path');
const core = require('@babel/core');
const dce = require('babel-plugin-remove-unused-vars');
const plugin = require('../');

expect.addSnapshotSerializer({
  test: value => value && typeof value.cssZero === 'object',
  print: ({cssZero}) => cssZero.toString(),
});

const transpile = file =>
  core.transformFileSync(path.resolve(__dirname, file), {
    plugins: [plugin, dce],
    babelrc: false,
  });

it.each([
  ['simple.jsx'],
  ['combining.jsx'],
  ['merging.jsx'],
  ['conditional.jsx'],
  ['conditional-with-dce.jsx'],
  ['pseudo.jsx'],
  ['media-query.jsx'],
  ['nested.jsx'],
])('%s', file => {
  const {code, metadata} = transpile(`./fixtures/${file}`);
  expect(code).toMatchSnapshot();
  expect(metadata).toMatchSnapshot();
});
