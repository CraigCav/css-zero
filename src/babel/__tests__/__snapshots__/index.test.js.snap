// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`combining.jsx 1`] = `"export const Component = () => <div className={\\"x1dqz7z3 x1e4w2a9\\"} />;"`;

exports[`combining.jsx 2`] = `
.x1dqz7z3 {color:red}
.x1e4w2a9 {font-size:16px}
`;

exports[`conditional.jsx 1`] = `
"export const LogicalAndExpression = props => <div className={(props.isBlue ? \\"x1vong5g\\" : \\"x1dqz7z3\\") + \\" \\" + \\"x1e4w2a9\\"} />;
export const LogicalExpressionDeterministic = () => <div className={\\"x1vong5g\\"} />;
export const IgnoreFalsey = () => <div className={\\"\\"} />;
export const SimpleTernaryExpression = () => <div className={\\"x1dqz7z3 x1e4w2a9\\"} />;"
`;

exports[`conditional.jsx 2`] = `
.x1vong5g {color:blue}
.x1dqz7z3 {color:red}
.x1e4w2a9 {font-size:16px}
`;

exports[`conditional-with-dce.jsx 1`] = `"export const LogicalAndExpression = props => <div className={(props.isBlue ? \\"x1vong5g\\" : \\"x1dqz7z3\\") + \\" \\" + \\"x1e4w2a9\\"} />;"`;

exports[`conditional-with-dce.jsx 2`] = `
.x1vong5g {color:blue}
.x1dqz7z3 {color:red}
.x1e4w2a9 {font-size:16px}
`;

exports[`media-query.jsx 1`] = `
"export const Component = () => <div className={\\"x1dqz7z3 x1lrxh9v\\"} />;
export const ComponentWithDuplicateMediaStyle = () => <div className={\\"x1ug2cuz xj95doo\\"} />;
export const ComponentWithOverriddenMediaStyle = () => <div className={\\"x17bvl1g xkuztku\\"} />;
export const ComponentWithNestedPseudo = () => <div className={\\"xfxv4bd x1oikiae\\"} />;
export const ComponentWithParentPseudo = () => <div className={\\"xashdej x1dp08z\\"} />;"
`;

exports[`media-query.jsx 2`] = `
.x1dqz7z3 {color:red}
.x1lrxh9v {@media screen and (min-width: 678px){color:green}}
.x1ug2cuz {color:black}
.xj95doo {@media screen and (min-width: 678px){color:yellow}}
.x17bvl1g {color:brown}
.xkuztku {@media screen and (min-width: 678px){color:salmon}}
.xfxv4bd {color:silver}
.x1oikiae:hover {@media screen and (min-width: 678px){color:gold}}
.xashdej {color:wheat}
.x1dp08z:hover {@media screen and (min-width: 678px){color:sandybrown}}
`;

exports[`merging.jsx 1`] = `"export const Component = () => <div className={\\"xp5623v\\"} />;"`;

exports[`merging.jsx 2`] = `.xp5623v {color:green}`;

exports[`nested.jsx 1`] = `"export const Component = () => <div className={\\"x194k0cs x1c2oj3k x1pvsa1a xz7a8fm xt08vo6\\"} />;"`;

exports[`nested.jsx 2`] = `
.x194k0cs h1 {font-size:32px}
.x1c2oj3k section {font-size:16px}
.x1pvsa1a section h1 {font-size:28px}
.xz7a8fm section {@media screen and (min-width: 678px){font-size:18px}}
.xt08vo6 section h1 {@media screen and (min-width: 678px){font-size:42px}}
`;

exports[`pseudo.jsx 1`] = `
"export const Component = () => <div className={\\"x1dqz7z3 x1jz5mb\\"} />;
export const ComponentWithDuplicatePseudoStyle = () => <div className={\\"x1ug2cuz x17vggq8\\"} />;
export const ComponentWithOverriddenPseudoStyle = () => <div className={\\"x17bvl1g xjhh7o6\\"} />;"
`;

exports[`pseudo.jsx 2`] = `
.x1dqz7z3 {color:red}
.x1jz5mb:hover {color:green}
.x1ug2cuz {color:black}
.x17vggq8:hover {color:yellow}
.x17bvl1g {color:brown}
.xjhh7o6:hover {color:salmon}
`;

exports[`simple.jsx 1`] = `"export const Component = () => <div className={\\"x1e4w2a9\\"} />;"`;

exports[`simple.jsx 2`] = `.x1e4w2a9 {font-size:16px}`;
